### PR TITLE
Add and fix mouse events on timeline

### LINKED ISSUE #226: Add an 'onHover' event to the timeline
Hi,

I'm very impressed with this library and have used it for 2 projects.

I've developed very easily a popup based on that you allowed HTML to be added in as content.

I'd like to know if you have any plans to make popups available to bind to hover over events.

Thanks,
Aaron

### LINKED ISSUE #2421: doubleClick Events it not fired.
when "clickToUse" is true once i clicked item i could use click event again i click the same item i could not use click event it not fired.

### PR DESCRIPTION
**Pull request by yotamberk**
fixes #226, #2421 and #2429

### CODE REVIEW

File: examples/timeline/interaction/eventListeners.html [hunk 1]
```diff
@@ -57,6 +57,36 @@
...
+
+    timeline.on('contextmenu', function (properties) {
+      logEvent('contextmenu', properties);
+    });
+
+    // other possible events:
+    // timeline.on('mouseDown', function (properties) {
+    //   logEvent('mouseDown', properties);
+    // });
+
```
Comment (mojoaxel) [inline examples/timeline/interaction/eventListeners.html]: empty line

File: lib/timeline/Timeline.js [hunk 2]
```diff
@@ -120,15 +120,27 @@ function Timeline (container, items, groups, options) {
   this.itemsData = null;      // DataSet
   this.groupsData = null;     // DataSet
 
-  this.on('tap', function (event) {
```
Comment (mojoaxel) [inline lib/timeline/Timeline.js]: Did you make sure this still workes on mobile?

File: docs/timeline/index.html [hunk 3]
```diff
@@ -1480,6 +1480,42 @@ <h2 id="Events">Events</h2>
     </tr>
 
     <tr>
+      <td>mouseOver</td>
+      <td>
+        Passes a properties object as returned by the method <a href="#getEventProperties"><code>Timeline.getEventProperties(event)</code></a>.
+      </td>
+      <td>Fired when mouse over the Timeline.
```
Comment (mojoaxel) [inline docs/timeline/index.html]: Better: "Fired when the mouse hovers over a timeline element."

File: docs/timeline/index.html [hunk 4]
```diff
@@ -1480,6 +1480,42 @@ <h2 id="Events">Events</h2>
...
+      </td>
+    </tr>
+
+    <tr>
+      <td>mouseDown</td>
+      <td>
+        Passes a properties object as returned by the method <a href="#getEventProperties"><code>Timeline.getEventProperties(event)</code></a>.
+      </td>
+      <td>Fired when mouse down inside the Timeline.
```
Comment (mojoaxel) [inline docs/timeline/index.html]: I don't understand this event. I only managed to trigger this in the example by pressing the middle-mouse-button.

File: docs/timeline/index.html [hunk 5]
```diff
@@ -1480,6 +1480,42 @@ <h2 id="Events">Events</h2>
...
+      </td>
+    </tr>
+
+    <tr>
+      <td>mouseUp</td>
```
Comment (mojoaxel) [inline docs/timeline/index.html]: I don't understand this event. I only managed to trigger this in the example by pressing the middle-mouse-button.

File: lib/timeline/Timeline.js [hunk 6]
```diff
@@ -120,15 +120,27 @@ function Timeline (container, items, groups, options) {
...
+  this.dom.root.onmousedown = function (event) {
+    me.emit('mouseDown', me.getEventProperties(event))
+  };
+  this.dom.root.onmouseover = function (event) {
+    me.emit('mouseOver', me.getEventProperties(event))
+  };
+  this.dom.root.onmouseup = function (event) {
+    me.emit('mouseUp', me.getEventProperties(event))
+  };
+  this.dom.root.onmousemove = function (event) {
```
Comment (mojoaxel) [inline lib/timeline/Timeline.js]: do we really want to emit this event. This happens quite often? What is the usecase for this?

File: lib/timeline/Timeline.js (hunk 2 above)
Comment (yotamberk) [inline lib/timeline/Timeline.js]: Yup, I have a touch screen on my computer and it seems to work fine.

File: docs/timeline/index.html (hunk 4 above)
Comment (yotamberk) [inline docs/timeline/index.html]: It fires when you press the left mouse button in the timeline

File: docs/timeline/index.html (hunk 5 above)
Comment (yotamberk) [inline docs/timeline/index.html]: Same as above just for when you release it

File: lib/timeline/Timeline.js (hunk 6 above)
Comment (yotamberk) [inline lib/timeline/Timeline.js]: I see a use for it when someone wants a tooltip appearing any time you're on the timeline and want to see the exact time you are hovering on

File: docs/timeline/index.html (hunk 3 above)
Comment (mojoaxel) [inline docs/timeline/index.html]: No problem. But the event is not fired by mouse over the "timeline" itself, but the event is fired if the mouse is over a "timeline element". Check it again, I'm not so sure myself but it looked like the event is e.g. fired by mouse over time-ticks, elements. Maybe we should clarify this!?

File: docs/timeline/index.html (hunk 4 above)
Comment (mojoaxel) [inline docs/timeline/index.html]: I get a "click" event on the left button and a "contextmenu" on the right button.

File: lib/timeline/Timeline.js (hunk 2 above)
Comment (mojoaxel) [inline lib/timeline/Timeline.js]: cool! But we should test it on mobile devices. It's possible that your screen emulates mouse events. I'll try to have a look on some devices.

Comment (yotamberk) [inline lib/timeline/Timeline.js]: Np

File: docs/timeline/index.html (hunk 3 above)
Comment (yotamberk) [inline docs/timeline/index.html]: Will do. I think ill change it to mouseover for the timeline as a  whole

File: docs/timeline/index.html (hunk 4 above)
Comment (yotamberk) [inline docs/timeline/index.html]: Weird... ill check that out

Comment (yotamberk) [inline docs/timeline/index.html]: I've removed this event. I see absolutely no real reason for this event.

File: docs/timeline/index.html (hunk 5 above)
Comment (yotamberk) [inline docs/timeline/index.html]: I've removed this event. I see absolutely no real reason for this event.

File: lib/timeline/Timeline.js (hunk 2 above)
Comment (mojoaxel) [inline lib/timeline/Timeline.js]: Ok, looks good on Android and iOS.

**Pull request merged**